crates/ui/src/sidebar/menu.rs

```diff
@@ -66,8 +67,9 @@ pub struct SidebarMenuItem {
 
 impl SidebarMenuItem {
     /// Create a new SidebarMenuItem with a label
-    pub fn new(label: impl Into<SharedString>) -> Self {
+    pub fn new(id: impl Into<ElementId>, label: impl Into<SharedString>) -> Self {
```

Comment:
Here not add to add `id` argument, we can set the id from parent menu iter with the index.

You can check the same thing in TabBar and Tab:

https://github.com/longbridge/gpui-component/blob/main/crates/ui/src/tab/tab.rs#L447-L464

https://github.com/longbridge/gpui-component/blob/main/crates/ui/src/tab/tab_bar.rs#L257-L263

Reply:
You're absolutely right. Thanks for the suggestion. I applied the changes. For the story, I had to make dynamic IDs that were different from each other, as using just the enumerated `usize` made it unreliable as it still did not work properly.